lib/json/pure/generator.rb

```diff
@@ -286,12 +288,70 @@ def to_h
         # created this method raises a
         # GeneratorError exception.
         def generate(obj)
-          result = obj.to_json(self)
+          if @indent.empty? and @space.empty? and @space_before.empty? and @object_nl.empty? and @array_nl.empty? and
+              !@ascii_only and !@script_safe and @max_nesting == 0 and !@strict
```

Comment:
This feels like this could be precomputed in `configure` so you'd only need to check a single ivar, no?

Reply:
Mmh, I'm not sure what's the protocol, the State class has a bunch of public `attr_accessor` so it looks like `configure` is not necessarily called after such accessors would be used to write.
Or maybe JSON users are not supposed to call these accessors? What are they for then?

BTW it seems pretty inefficient that e.g. JSON.generate would create a new State every time (which runs the State constructor and configure every time) instead of using `SAFE_STATE_PROTOTYPE.dup` or so, but that's a separate thing so improve later.

Reply:
Out of curiosity I did 
```patch
diff --git a/lib/json/pure/generator.rb b/lib/json/pure/generator.rb
index 9f44ef5..d07c2de 100644
--- a/lib/json/pure/generator.rb
+++ b/lib/json/pure/generator.rb
@@ -268,6 +268,10 @@ module JSON
           else
             @max_nesting = 0
           end
+
+          @fast = (@indent.empty? and @space.empty? and @space_before.empty? and @object_nl.empty? and @array_nl.empty? and
+            !@ascii_only and !@script_safe and @max_nesting == 0 and !@strict)
+
           self
         end
         alias merge configure
@@ -277,6 +281,7 @@ module JSON
         def to_h
           result = {}
           instance_variables.each do |iv|
+            next if iv == :@fast
             iv = iv.to_s[1..-1]
             result[iv.to_sym] = self[iv]
           end
@@ -292,8 +297,10 @@ module JSON
         def generate(obj)
           if @indent.empty? and @space.empty? and @space_before.empty? and @object_nl.empty? and @array_nl.empty? and
               !@ascii_only and !@script_safe and @max_nesting == 0 and !@strict
+            raise "OOPS" if !@fast
             result = generate_json(obj, '')
           else
+            raise "OOPS" if @fast
             result = obj.to_json(self)
           end
           JSON.valid_utf8?(result) or raise GeneratorError,
```
To see if there were any inconsistency and the test suite didn't find any.
But still, I think we need to understand better how these accessors are used (or maybe remove them if they're not needed).